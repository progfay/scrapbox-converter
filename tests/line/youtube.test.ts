/* global describe it expect */

import { parse } from '@progfay/scrapbox-parser'
import convert from '../../src'
import pretty from 'pretty'

describe('youtube', () => {
  it('Simple youtube', () => {
    const input = `Youtube
[https://www.youtube.com/watch?v=YJ5q8Wrkbdw]`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })

  it('Youtube with start time', () => {
    const input = `Youtube
[https://www.youtube.com/watch?v=YJ5q8Wrkbdw&t=8]
[https://www.youtube.com/watch?v=YJ5q8Wrkbdw&t=0h8s]`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })

  it('Short link youtube', () => {
    const input = `Youtube
[https://youtu.be/YJ5q8Wrkbdw]`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })

  it('Short link youtube with start time', () => {
    const input = `Youtube
[https://youtu.be/YJ5q8Wrkbdw?t=5]
[https://youtu.be/YJ5q8Wrkbdw?t=1m]`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })

  it('Invalid start time', () => {
    const input = `Youtube
[https://www.youtube.com/watch?v=YJ5q8Wrkbdw&t=hoge]`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })
})
