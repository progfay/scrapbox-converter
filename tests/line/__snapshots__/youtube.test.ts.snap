// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`youtube Invalid start time 1`] = `
"<article>
  <h1 class=\\"title\\">Youtube</h1>
  <section>
    <p>
      <amp-youtube data-videoid=\\"YJ5q8Wrkbdw\\" layout=\\"responsive\\" width=\\"480\\" height=\\"270\\" data-param-start=\\"0\\"></amp-youtube>
    </p>
  </section>
</article>"
`;

exports[`youtube Short link youtube 1`] = `
"<article>
  <h1 class=\\"title\\">Youtube</h1>
  <section>
    <p>
      <amp-youtube data-videoid=\\"YJ5q8Wrkbdw\\" layout=\\"responsive\\" width=\\"480\\" height=\\"270\\" data-param-start=\\"0\\"></amp-youtube>
    </p>
  </section>
</article>"
`;

exports[`youtube Short link youtube with start time 1`] = `
"<article>
  <h1 class=\\"title\\">Youtube</h1>
  <section>
    <p>
      <amp-youtube data-videoid=\\"YJ5q8Wrkbdw\\" layout=\\"responsive\\" width=\\"480\\" height=\\"270\\" data-param-start=\\"5\\"></amp-youtube>
    </p>
    <p>
      <amp-youtube data-videoid=\\"YJ5q8Wrkbdw\\" layout=\\"responsive\\" width=\\"480\\" height=\\"270\\" data-param-start=\\"60\\"></amp-youtube>
    </p>
  </section>
</article>"
`;

exports[`youtube Simple youtube 1`] = `
"<article>
  <h1 class=\\"title\\">Youtube</h1>
  <section>
    <p>
      <amp-youtube data-videoid=\\"YJ5q8Wrkbdw\\" layout=\\"responsive\\" width=\\"480\\" height=\\"270\\" data-param-start=\\"0\\"></amp-youtube>
    </p>
  </section>
</article>"
`;

exports[`youtube Youtube with start time 1`] = `
"<article>
  <h1 class=\\"title\\">Youtube</h1>
  <section>
    <p>
      <amp-youtube data-videoid=\\"YJ5q8Wrkbdw\\" layout=\\"responsive\\" width=\\"480\\" height=\\"270\\" data-param-start=\\"8\\"></amp-youtube>
    </p>
    <p>
      <amp-youtube data-videoid=\\"YJ5q8Wrkbdw\\" layout=\\"responsive\\" width=\\"480\\" height=\\"270\\" data-param-start=\\"8\\"></amp-youtube>
    </p>
  </section>
</article>"
`;
